templates/go/duration.go

```diff
@@ -4,9 +4,31 @@ const durationTpl = `{{ $f := .Field }}{{ $r := .Rules }}
 	{{ template "required" . }}
 
 	{{ if or $r.In $r.NotIn $r.Lt $r.Lte $r.Gt $r.Gte $r.Const }}
+		{{ if gogo }}
+			{{ if .Gogo.Stdduration }}
+				{{ if .Gogo.Nullable }}
+		if d := {{ accessor . }}; d != nil {
+			dur := *d
+				{{ else }}
+		if true {
+			dur := {{ accessor . }}
+				{{ end }}
+			{{ else }}
+				{{ if .Gogo.Nullable }}
+		if d := {{ accessor . }}; d != nil {
+			dur, err := types.DurationFromProto(d)
+			if err != nil { return {{ errCause . "err" "value is not a valid duration" }} }
+				{{ else }}
+		if d := {{ accessor . }}; true {
+			dur, err := types.DurationFromProto(&d)
+			if err != nil { return {{ errCause . "err" "value is not a valid duration" }} }
+				{{ end }}
+			{{ end }}
+		{{ else }}
```

Comment:
Instead of these big conditionals, how about we move the shared templates into something like `goshared` and have separate templates for where the two targets diverge? So everything but `duration`, `time`, and `file` (or perhaps just a `preamble`) are shared. Does that make sense?

Reply:
Done